verify/cert.go

```diff
@@ -166,3 +190,47 @@ func (cv *CertVerifier) retrieveBatchMetadataHash(ctx context.Context, batchID u
 	}
 	return onchainHash, nil
 }
+
+// getQuorumParametersAtLatestBlock fetches the required quorums and quorum adversary thresholds
+// from the EigenDAServiceManager contract at the latest block.
+// We then cache these parameters and use them in the Verifier to verify the certificates.
+//
+// Note: this strategy (fetching once and caching) only works because these parameters are immutable.
+// They might be different in different environments (for eg on a devnet or testnet), but they are fixed on a given network.
+// We used to allow these parameters to change (via a setter function on the contract), but that then forced us here in the proxy
+// to query for these parameters on every request, at the batch's reference block number (RBN).
+// This in turn required rollup validators running this proxy to have an archive node, in case the RBN was >128 blocks in the past,
+// which was not ideal. So we decided to make these parameters immutable, and cache them here.
+func getQuorumParametersAtLatestBlock(
+	manager *binding.ContractEigenDAServiceManagerCaller,
+) ([]uint8, map[uint8]uint8, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
How did you settle on 5 seconds? Would it be worth making configurable? (maybe not considering imminent v1 deprecation)

Reply:
Arbitrary number that I typically use everywhere for calls (as its ample time to request an eth_call even from a node that halfway across the globe). Indeed don't think its worth creating a flag for this given v1 deprecation.

Overall this is another thing that I would to put in the style guide. Not sure if you have an opinion about this, some people don't like for constructors to make network calls. because what I should be doing is adding a context to the constructor to hint at this, and let call sites pass in the context that they want. But looking quickly that just bubbled up like 4 levels of constructors... so not sure what the right strategy is here, and don't really want to spend too much time on this atm, unless you have a good idea.

Reply:
I don't have a great idea for this - no complaints leaving it as-is